### ISSUE TITLE
Dragging an item as it's removed from the layout causes funny behaviour

### ISSUE THREAD

**Issue by DanielLeone** [react-grid-layout/react-grid-layout#1836]
### Describe the bug

If you're dragging a Grid Item as it's removed from the layout prop, the `placeholder` gets stuck in place, and unresponsive, until you pick up a different grid item.

### Your Example Website or App

https://codesandbox.io/s/react-grid-layout-disappearing-drag-item-6fpjis

### Steps to Reproduce the Bug or Issue

1. Click the "Remove Item B in 1s" button
2. Start dragging item B immediately, before it gets removed
3. Wait for item B to be removed from the layout
4. Observe the placeholder remaining around

### Expected behavior

As a user, I expect the dragging item to just disappear, maybe the layout be rearranged a bit, as through I dragged the item out of the grid altogether.

### react-grid-layout library version

1.3.4

### Operating System Version

macOS

### Browser

Chrome

### Additional context

Sometimes I've noticed the dragging item gets swapped to a different grid item, as it's removed, maybe a different issue.

I'm guessing the solution might be something like the `onDragLeave` event, ie just call `this.removeDroppingPlaceholder()`, but at the appropriate time - which is something like, when the `layout` prop or React.Children changes (I'm not sure what's correct)....

Basically, layout property and children array get updated to remove the item the user is currently dragging around. We need to somehow clear our the dragging item if that happens I think.

**Comment by DanielLeone** [comment]
Can confirm this patch fixes the issue for me, although I'm unsure if it's the best spot for the code; maybe `getDerivedStateFromProps` would be better, any suggestions? 👍🙏

```diff
Index: lib/ReactGridLayout.jsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/ReactGridLayout.jsx b/lib/ReactGridLayout.jsx
--- a/lib/ReactGridLayout.jsx	(revision 30ac6508c2d422a425f7aa2742ccb889b9153e0d)
+++ b/lib/ReactGridLayout.jsx	(date 1674439651653)
@@ -206,6 +206,16 @@
   }
 
   componentDidUpdate(prevProps: Props, prevState: State) {
+    if (this.state.activeDrag) {
+      const children = this.props.children || [];
+      const draggingKey = this.state.activeDrag.i;
+      if (!children.find(child => child.key === draggingKey)) {
+        // the item we're dragging around has been removed from children,
+        //  so we need to clean up our state and forget about it
+        this.removeDroppingPlaceholder()
+      }
+    }
+
     if (!this.state.activeDrag) {
       const newLayout = this.state.layout;
       const oldLayout = prevState.layout;

```

**Comment by STRML** [comment]
Closing as this is an edge case that should be handled in v2's improved state management.

If you're removing items while they're being dragged, consider disabling drag during removal or using React's key prop to force unmount.

Please test with v2.2.0+ and reopen if the stuck placeholder issue persists.
